src/FieldWarning/Assets/Units/Waypoint/PathfinderData.cs

```diff
@@ -133,8 +131,9 @@ public void RemoveArc(PathNode node1, PathNode node2)
     // Gives the relative speed of a unit with the given MobilityType at the given location
     // Relative speed is 0 if the terrain is impassible and 1 for road, otherwise between 0 and 1
     // If radius > 0, check for units in the way, otherwise just look at terrain
-    public float GetUnitSpeed(MobilityType mobility, Vector3 location, float radius)
+    public float GetUnitSpeed(MobilityType mobility, Vector3 location, float radius, Vector3 direction)
```

Comment:
I realize this is mostly old code, but I spent a couple minutes wondering what "radius" is. Rename to "unitRadius"? I think the code can be made more readable by moving the first loop into a method of its own (if the purpose of the method is to avoid collisions, we name it (for a bad example) "IsSomethingInTheWay" and this method does `if (IsSomethingInTheWay(args)) return;`. Just some food for thought, I am not familiar with the pathfinder code base.

Reply:
I just now realized these comments were here.

You can make whatever changes you want to the code, of course. At this stage I just assume that all the code will need to change some.